internal/pkg/services/load-balancer/utils/utils.go

```diff
@@ -34,13 +34,23 @@ func GetLoadBalancerTargetPool(ctx context.Context, apiClient LoadBalancerClient
 		return nil, fmt.Errorf("no target pools found")
 	}
 
-	for _, targetPool := range *resp.TargetPools {
+	targetPool := FindLoadBalancerTargetPoolByName(resp.TargetPools, targetPoolName)
+	if targetPool == nil {
+		return nil, fmt.Errorf("target pool not found")
+	}
+	return targetPool, nil
+}
+
+func FindLoadBalancerTargetPoolByName(targetPools *[]loadbalancer.TargetPool, targetPoolName string) *loadbalancer.TargetPool {
```

Comment:
I think the function signature doesn't need to take a `*[]loadbalancer.TargetPool`, the argument could simply be a `[]loadbalancer.TargetPool`